device/cuda/include/traccc/cuda/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.hpp

```diff
@@ -0,0 +1,72 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Local include(s).
+#include "traccc/cuda/utils/stream.hpp"
+
+// Project include(s).
+#include "traccc/ambiguity_resolution/ambiguity_resolution_config.hpp"
+#include "traccc/edm/track_candidate.hpp"
+#include "traccc/utils/algorithm.hpp"
+#include "traccc/utils/memory_resource.hpp"
+#include "traccc/utils/messaging.hpp"
+
+// VecMem include(s).
+#include <vecmem/utils/copy.hpp>
+
+namespace traccc::cuda {
+
+/// Evicts tracks that seem to be duplicates or fakes. This algorithm takes a
+/// greedy approach in the sense that it will remove the track which looks "most
+/// duplicate/fake"
+class greedy_ambiguity_resolution_algorithm
+    : public algorithm<track_candidate_container_types::buffer(
+          const typename track_candidate_container_types::const_view&)>,
```

Comment:
You're a bit too liberal with `typename`. :stuck_out_tongue: It's needed when referring to a "sub-type" of a template type. But here `track_candidate_container_types::const_view` is a concrete type.

I.e.:

```suggestion
          const track_candidate_container_types::const_view&)>,
```

Notice how you also didn't use `typename` on line 53. :wink: